### PR TITLE
add basic scaffold for web app

### PR DESCRIPTION
**Pull request by butlerx**
add stimulus for clientside app
rocket for api and file serving 
handlebar templating
webpack to build js and scss

### CODE REVIEW

**Comment by GoldenBadger** [comment]
What does one need to do to build and run this?

**Comment by butlerx** [comment]
I've added build instructions

this also won't build on stable or beta due to rocket

**Comment by GoldenBadger** [comment]
Can you rebase and squash this yourself to make the commit history less weird? :P

**Comment by butlerx** [comment]
cleaner?

**Comment by GoldenBadger** [comment]
The build instructions give this error:

```
     Running `target/debug/good_stv --server`
error: The following required arguments were not provided:
    <seats>

USAGE:
    good_stv <seats> --server

For more information try --help
```

**Comment by GoldenBadger** [comment]
I believe in clap you can group mutually exclusive options. So if `--server` is passed it ignores all other arguments which are otherwise required.

**Comment by butlerx** [comment]
I've been looking at the doc and i can't seem to find it

**Comment by GoldenBadger** [comment]
I believe this will do what you want: https://docs.rs/clap/2.31.1/clap/struct.ArgGroup.html

**Comment by butlerx** [comment]
that's exactly what I'm looking for, thanks.

File: src/main.rs [hunk 1]
```diff
@@ -14,23 +14,36 @@
...
+#![feature(custom_attribute)]
+#![feature(use_extern_macros)]
+
 extern crate clap;
 extern crate csv;
 extern crate env_logger;
 #[macro_use]
 extern crate failure;
-#[macro_use]
+#[macro_use(log, info, debug)]
```
Comment (GoldenBadger) [inline src/main.rs]: I think this should be replaced by just `#[macro_use(log)]` and then all logging macros are called as `log::error!` etc.

File: src/main.rs [hunk 2]
```diff
@@ -14,23 +14,36 @@
...
+extern crate serde_derive;
 
 mod stv;
+mod routes;
 
 use std::io;
 
-use clap::{App, Arg};
+use rocket_contrib::Template;
+
```
Comment (GoldenBadger) [inline src/main.rs]: Nit: remove this newline and sort these alphabetically.

File: src/main.rs [hunk 3]
```diff
@@ -14,23 +14,36 @@
...
+use clap::{App, Arg, ArgGroup};
 
 use failure::{Error, ResultExt};
 use stv::*;
+use routes::*;
```
Comment (GoldenBadger) [inline src/main.rs]: Nit: sort these alphabetically

File: src/main.rs (hunk 1 above)
Comment (GoldenBadger) [reply src/main.rs]: See https://github.com/SergioBenitez/Rocket/issues/109#issuecomment-300326193 for more info.

Comment (butlerx) [reply src/main.rs]: I attempted this and got compiler errors in `std.rs` 
```text
failed to resolve. Not a module `log`
```
its possible I'm missing something

Comment (butlerx) [reply src/main.rs]: i now see what i was missing

**Pull request merged**